### PR TITLE
Update line color selection in muon analysis

### LINKED ISSUE #29064: Muon line colours
In Muon Analysis the plots should always follow the same line colour pattern. 

So if you create a new plot with 3 lines, you will always get the same 3 colours.

### PR DESCRIPTION
**Pull request by Matthew-Andrew**
**Description of work.**

This tracks the currently plotted colors in the plots in the MuonAnalysis Gui to ensure we have a consistent set of colours aw the figure is replotted and workspaces are added and removed.

**To test:**

* Try using the GUI and producing plots and check that the line colouring remains clear and consistent.
 * Note that #29076 already exists on master and is not resolved here.

Fixes #29064  

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by AnthonyLim23** [comment]
I have had a think about this over the weekend. Could you make sure there is a way to get the next colour and then force multiple lines to have a specified colour? This is something we are going to want when adding vertical lines for elemental analysis

**Comment by Mantid-Matthew** [comment]
> I have had a think about this over the weekend. Could you make sure there is a way to get the next colour and then force multiple lines to have a specified colour? This is something we are going to want when adding vertical lines for elemental analysis

I think this may be scope creep on this current PR. When we add the ability to put vertical lines on the plot this will most likely require adding new methods to the view interface. It should be relatively easy to within these new methods handle all vertical lines being one color.

**Comment by AnthonyLim23** [comment]
thats all I want. Just want to make sure we dont have to rewrite a significant part of this to allow it

File: scripts/Muon/GUI/Common/plot_widget/plotting_canvas/plotting_canvas_view.py [hunk 1]
```diff
@@ -85,6 +87,9 @@ def create_new_plot_canvas(self, num_axes):
         """Creates a new blank plotting canvas"""
         self._plot_information_list = []
         self._number_of_axes = num_axes
+        self._color_queue = [ColorQueue()]
```
Comment (StephenSmith25) [inline scripts/Muon/GUI/Common/plot_widget/plotting_canvas/plotting_canvas_view.py:90]: I think we may end up with one more ColorQueue object than necessary here, as we initialize the list with one object and then loop over and append num_axes more. Maybe replacing the loop with something like 
```suggestion
        self._color_queue = [ColorQueue() for _ in range(num_axes) ]
```
would suffice.

**Review by StephenSmith25 — COMMENTED** [review]
Looks good to me, colors stay consistent throughout everything I tested. Just one small comments and a docs merge conflict to sort out.

File: scripts/Muon/GUI/Common/plot_widget/plotting_canvas/plot_color_queue.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+import heapq
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/plot_widget/plotting_canvas/plot_color_queue.py:7]: can we have a comment with a link to the doc for this? I dont think it is common knowledge

**Comment by AnthonyLim23** [comment]
we have a conflict

**Review by StephenSmith25 — APPROVED** [review]
Looks good to go 👍

**Pull request merged**